src/kvstore/Part.cpp

```diff
@@ -253,6 +255,30 @@ bool Part::commitLogs(std::unique_ptr<LogIterator> iter) {
     return engine_->commitBatchWrite(std::move(batch)) == ResultCode::SUCCEEDED;
 }
 
+std::pair<int64_t, int64_t> Part::commitSnapshot(const std::vector<std::string>& rows,
+                                                 LogID committedLogId,
+                                                 TermID committedLogTerm,
+                                                 bool finished) {
+    auto batch = engine_->startBatchWrite();
+    int64_t count = 0;
+    int64_t size = 0;
+    for (auto& row : rows) {
+        count++;
+        size += row.size();
+        auto kv = decodeKV(row);
+        CHECK_EQ(ResultCode::SUCCEEDED, batch->put(kv.first, kv.second));
```

Comment:
maybe need change use too many micro CHECK, it is not safe crash directly.

Reply:
yep. We are replace it with check status :)

Reply:
Good point.